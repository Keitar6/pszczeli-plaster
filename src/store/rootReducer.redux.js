import { combineReducers } from "redux";

import { userReducer } from "./userReducer/user.reducer";
import { generalPropReducer } from "./generalPropReducer/generalProp.reducer";
import { cartReducer } from "./cartReducer/cart.reducer";

const rootReducer = combineReducers({
  userReducer,
  generalPropReducer,
  cartReducer,
});
export default rootReducer;
