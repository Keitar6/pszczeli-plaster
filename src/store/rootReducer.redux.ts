import { combineReducers } from "redux";

import { userReducer } from "./userReducer/user.reducer";
import { generalPropReducer } from "./generalPropReducer/generalProp.reducer";
import { cartReducer } from "./cartReducer/cart.reducer";
import { categoriesReducer } from "./categories/category.reducer";
import { orderHistoryReducer } from "./orderHistory/orderHistory.reducer";

export type ReduxState = ReturnType<typeof rootReducer>;

const rootReducer = combineReducers({
  userReducer,
  generalPropReducer,
  cartReducer,
  categoriesReducer,
  orderHistoryReducer
});
export default rootReducer;
