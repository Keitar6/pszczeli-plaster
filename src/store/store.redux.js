import { compose, createStore, applyMiddleware } from "redux";
import logger from "redux-logger";
import rootReducer from "./rootReducer.redux";

const middleWares = [logger];

const composedEnhancer = compose(applyMiddleware(...middleWares));

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
