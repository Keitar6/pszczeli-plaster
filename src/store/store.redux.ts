import { compose, createStore, applyMiddleware } from "redux";
import type { Middleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";

import rootReducer, { ReduxState } from "./rootReducer.redux";

import { persistStore, persistReducer } from "redux-persist";
import type { PersistConfig } from "redux-persist";
import storage from "redux-persist/lib/storage";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

type ExtendedPersistConfig = PersistConfig<ReduxState> & {
  whitelist: (keyof ReduxState)[];
};

const persistConfig: ExtendedPersistConfig = {
  key: "root",
  storage: storage, // local storage
  whitelist: ["cartReducer", "userReducer", "categoriesReducer"]
};

const middleWares = [
  process.env.NODE_ENV !== "production" && logger,
  thunk
].filter((middleware): middleware is Middleware => Boolean(middleware));

const composeEnhancer =
  (process.env.NODE_ENV !== "production" &&
    window &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;

const composedEnhancers = composeEnhancer(applyMiddleware(...middleWares));

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  undefined,
  composedEnhancers
);

export const persistor = persistStore(store);

export type AppDispatch = typeof store.dispatch;
