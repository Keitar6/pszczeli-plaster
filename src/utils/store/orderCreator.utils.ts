import { DeliveryData, Order, OrderItem } from "store/orderHistory/orderHistory.types";

export const orderCreator= (
  id: string,
  time: string,
  price: number,
  itemsBought: OrderItem[],//{,quantity:15}
  deliveryData: DeliveryData,
  deliveryPrice: number
):Order => {
  return {
    id,
    time,
    price,
    itemsBought,
    deliveryData,
    deliveryPrice
  };
};
