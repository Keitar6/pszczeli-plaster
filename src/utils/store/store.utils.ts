import type { AnyAction } from "redux";

export type ActionWithPayload<T, P> = { type: T; payload: P };

export type Action<T> = { type: T };

export type SomeAction<T, P> = Action<T> | ActionWithPayload<T, P>;

export function actionCreator<T extends string, P>(
  type: T,
  payload: P
): ActionWithPayload<T, P>;

export function actionCreator<T extends string>(
  type: T,
  payload: void
): Action<T>;

export function actionCreator<T extends string, P>(type: T, payload: P) {
  return {
    type,
    payload
  };
}
/*
Funkcja dodająca do funkcji zwracającej actionCreator funkcję sprawdzającą(match(action)) czy 
action otrzymany przez reducer się zgadza
*/
type Matchable<AC extends () => AnyAction & { type: string }> = AC & {
  type: ReturnType<AC>["type"];
  match(action: AnyAction): action is ReturnType<AC>;
};

export function withMatch<
  AC extends (...args: any[]) => AnyAction & { type: string }
>(actionCreator: AC): Matchable<AC>;

export function withMatch<AC extends () => AnyAction & { type: string }>(
  actionCreator: AC
): Matchable<AC>;

export function withMatch<AC extends () => AnyAction & { type: string }>(
  actionCreator: AC
) {
  return Object.assign(actionCreator, {
    match(action: AnyAction) {
      return action.type === actionCreator().type;
    }
  });
}
