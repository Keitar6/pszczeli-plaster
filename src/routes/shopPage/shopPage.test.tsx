import ShopPage from "./shopPage.route";
import ComponentsRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { AppStore } from "../../store/store.redux";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch
}));

describe("shopPage", () => {
  const store = AppStore;
  test("shopPage - snapshot", () => {
    const shopPage = ComponentsRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <ShopPage />
        </BrowserRouter>
      </Provider>
    ).toJSON();
    expect(shopPage).toMatchSnapshot();
  });
});
