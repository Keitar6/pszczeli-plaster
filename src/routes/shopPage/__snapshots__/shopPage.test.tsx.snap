// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopPage shopPage - snapshot 1`] = `
.c3 {
  font-size: 3.05 rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0rem 0 0 0;
}

.c4 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 70%;
  -ms-flex: 1 1 70%;
  flex: 1 1 70%;
  gap: 2rem;
}

.c5 {
  border-bottom: 1px solid #0230473e;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 0 0.6rem 0rem;
}

.c5 H1 {
  width: 100%;
  color: #FFB703;
  text-align: center;
  text-transform: uppercase;
}

.c2 {
  border-right: 1px solid #0230473e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5rem;
  padding: 0 0 0.6rem 0rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  padding: 0.5rem 5rem 0.5rem 0;
  gap: 1rem;
}

.c10 {
  cursor: pointer;
}

.c7 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  gap: 0.5rem;
  font-family: "Tapestry","Dancing Script",cursive;
  font-size: 1 rem;
  font-weight: bold;
  background-color: transparent;
  border-radius: 5rem;
}

.c8 {
  min-width: 10rem;
  width: auto;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 1rem 3rem;
  font-size: 1 rem;
  background-color: #faeff0;
  font-family: "Open Sans Condensed";
  font-weight: bolder;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8:hover {
  background-color: #023047;
  color: #FFB703;
}

.c8:focus {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all 0.6s ease;
  transition: all 0.6s ease;
}

.c9 {
  font-family: "Tapestry","Dancing Script",cursive;
  font-size: 1 rem;
  font-weight: bold;
  min-width: 5rem;
  padding: 0.2rem 1rem;
  border: 2px solid #023047;
  border-radius: 5rem;
}

.c9:hover {
  color: #023047;
  background-color: #0230473e;
}

.c9:focus {
  background-color: #FFB703;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transition: none;
  transition: none;
}

.c12 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 0.5rem;
}

.c11 {
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 0.5rem;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      <span>
        Menu
      </span>
    </h1>
    <ul
      className=""
    />
  </div>
  <div
    className="c0 c4"
  >
    <div
      className="c5"
    >
      <h1
        className="c3"
      >
        sklep
      </h1>
      <div
        className="c6"
      >
        <div
          className="c0 c7"
          id="SortingTypes"
        >
          <button
            className="c8 c9"
            onClick={[Function]}
          >
            Sortowanie
          </button>
        </div>
        <div
          className="c10"
          data-testid="ascenIconOnClick"
          onClick={[Function]}
        >
          <span />
        </div>
      </div>
    </div>
    <div
      className="c0 c11"
    >
      <div
        className="c0 c12"
      />
    </div>
  </div>
</div>
`;
